lib/apiservers/engine/backends/filter/container.go

```diff
@@ -87,31 +84,59 @@ func IncludeContainer(listContext *ContainerListContext, container *models.Conta
 	}
 
 	// filter on network name
-	for n := range container.Endpoints {
-		name := container.Endpoints[n].Scope
-		if listContext.Filters.Include("network") {
-			err := listContext.Filters.WalkValues("network", func(value string) error {
-				if name == value {
-					return nil
-				}
-				return fmt.Errorf("not part of container network")
-			})
-			if err != nil {
-				return ExcludeAction
+	if listContext.Filters.Include("network") {
+		netFilterValues := listContext.Filters.Get("network")
+
+		// Gather the container's networks in a map
+		numNetworks := len(container.Endpoints)
```

Comment:
The code in its current form works as expected for the above cases. I've pushed a commit with another integration test. I'll refactor the map to use a list in a later commit.